src/Consolonia.Core/ConsoloniaLifetime.cs

```diff
@@ -194,9 +194,12 @@ protected virtual void Dispose(bool disposing)
                     // TODO: dispose managed state (managed objects)
                     _cts?.Dispose();
                     _cts = null;
-                    var consoleTopLevelImpl = (ConsoleWindowImpl)MainWindow.PlatformImpl;
-                    ArgumentNullException.ThrowIfNull(consoleTopLevelImpl, nameof(consoleTopLevelImpl));
-                    consoleTopLevelImpl.Dispose();
+                    if (MainWindow != null && MainWindow.PlatformImpl != null)
```

Comment:
is it correct fix? Why when project has not been disposed yet, it already does not have a Window or even more - native windows has gone?